src/dmd/backend/dwarfdbginf.d

```diff
@@ -1070,9 +1058,34 @@ static if (1)
 
             debug_line.initialize();
 
-            debugline = DebugLineHeader.init;
+            debug_line.buf.write32(0);              // unit length
```

Comment:
The reason to use struct DebugLineHeader is it is faster to write it as a memcpy rather than all the following individual writes.
Why this refactor?

Reply:
Same answer as for https://github.com/dlang/dmd/pull/12201#discussion_r579083401.